pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -36,9 +38,19 @@ func (n *NetworkPolicyController) addClusterGroup(curObj interface{}) {
 	newGroup := n.processClusterGroup(cg)
 	klog.V(2).Infof("Creating new internal Group %s with selector (%s)", newGroup.UID, newGroup.Selector.NormalizedName)
 	n.internalGroupStore.Create(newGroup)
-	if newGroup.IPBlock == nil {
-		n.enqueueInternalGroup(key)
+	if newGroup.IPBlock != nil {
+		// ClusterGroup phase will be set to Realized for IPBlock CG.
+		err := n.updateGroupStatus(cg.Name, corev1a2.GroupRealized)
```

Comment:
Would it be better to call external API in syncHandler? For two reasons:
1. EventHandler is executed synchronously, external API may block other ClusterGroup's handling, while external APIs for different ClusterGroups can be called in parallel if in SyncHandler.
2. There are no retries if there is a transient error with apiserver/network here. SyncHandler can retry without blocking other items.